lib/java/persistance/src/main/java/org/enso/persist/PerGenerator.java

```diff
@@ -140,16 +139,14 @@ private void dump(Logger log, int length) {
             return a.getValue()[0] - b.getValue()[0];
           });
 
-      log.fine("==== Top Bytes & Counts of Classes =====");
+      log.debug("==== Top Bytes & Counts of Classes =====");
       for (var i = 0; i < list.size(); i++) {
         if (i == 30) {
           break;
         }
         var elem = list.get(list.size() - 1 - i);
-        log.log(
-            Level.FINE,
-            "  {0} {1} {2}",
-            new Object[] {elem.getValue()[0], elem.getValue()[1], elem.getKey().getName()});
+        log.debug(
+            "  " + elem.getValue()[0] + " " + elem.getValue()[1] + " " + elem.getKey().getName());
```

Comment:
Unfortunately the `  {0} {1} {2}` format doesn't work with `sfl4j`. Sad outcome of using two different logging system throughout the project.

Reply:
I believe the correct way to do this in slf4j is just use empty brackets `{}` instead of `{0}`. As an example, look into https://github.com/enso-org/enso/blob/c1098865f2bf093d89f60fa981c73352d2d13f34/engine/language-server/src/main/java/org/enso/languageserver/boot/resource/RepoInitialization.java#L160

Reply:
Concatenating strings manually relies on standard Java mechanism rather than relying on the logging framework semantics. As it is guaranteed the message is going to be printed, I opted for the generic mechanism known to work.

Reply:
> It is fine for this case.

Yes, it is [absolutely fine in this case](https://github.com/enso-org/enso/pull/8881/files#r1470572989) as we know we are going to log.